source/developers-guide/shopware-5-upgrade-guide-for-developers/index.md

```diff
@@ -17,14 +17,205 @@ including minor and bugfix releases, refer to the `UPGRADE.md` file found in you
+- Updated `Mpdf` to 7.0.3
+- Updated `doctrine/common` to 2.7.3
+- Updated `beberlei/assert` to 2.9.2
+
+### Basket refactoring
+
+To improve basket performance and ease of use for plugin developers, the `sBasket` class was refactored slightly
+while still being compatible with existing plugins as much as possible.
+
+The most relevant change is in regards to the following blocks in `themes/Frontend/Bare/frontend/checkout/cart_item.tpl`
+which were moved to their own files:
+    * `frontend_checkout_cart_item_product`
+    * `frontend_checkout_cart_item_premium_product`
+    * `frontend_checkout_cart_item_voucher`
+    * `frontend_checkout_cart_item_rebate`
+    * `frontend_checkout_cart_item_surcharge_discount`
+
+These five blocks were moved to their own template files in `themes/Frontend/Bare/frontend/checkout/` to optimize the include process:
+    * `cart_item_product.tpl`
+    * `cart_item_premium_product.tpl`
+    * `cart_item_rebate.tpl`
+    * `cart_item_voucher.tpl`
+    * `cart_item_surcharge_discount.tpl`
+
+Also, the following templates no longer extend `cart_item.tpl` but include the logic themselves and have their own subtemplates:
+    * `confirm_item_premium_product.tpl`
+    * `confirm_item_product.tpl`
+    * `confirm_item_rebate.tpl`
+    * `confirm_item_surcharge_discount.tpl`
+    * `confirm_item_voucher.tpl`
+    * `finish_item_premium_product.tpl`
+    * `finish_item_product.tpl`
+    * `finish_item_voucher.tpl`
+
+If your theme extends one of the contained blocks, you'll have to change the filename it extends from.
+
+The following classes were added to simplify changes to the checkout process by plugins:
+* Added struct `Shopware\Components\Cart\Struct\CartItemStruct` to represent items in the cart during calculation
+* Added public function `sBasket::updateCartItems` to provide a new way of interacting with cart updates
+
+### Proportional tax calculation
+
+The proportional tax calculation allows to calculate multiple taxes for all fees, vouchers, discounts etc in baskets
+containing items with e.g. 19% and 7% tax. This feature needs to be activated in *Settings*, *Checkout*, *Proportional calculation of tax positions*.
+For the proportional tax calculation to work with vouchers and modes of dispatch, be sure to set the mode of tax calculation to "auto detection" in their settings
+
+The following were made to implement this feature:
+
+* Added `Shopware\Components\Cart\ProportionalTaxCalculator` to calculate proportional taxes for the cart items
+* Added `Shopware\Components\Cart\BasketHelper` to to add items to the cart that need to be calculation in a proportional way
+* Added `Shopware\Components\Cart\ProportionalCartMerger` to merge proportional cart items into one cart item
+* Added new filter event to modify proportional vouchers `Shopware_Modules_Basket_AddVoucher_VoucherPrices`   
+* Added new column `invoice_shipping_tax_rate` to s_order, to save exact dispatch shipping tax rate.
```

Comment:
new column `is_proportional_calculation` to s_order, which defines that the order is made with proportional items 😊

Reply:
Done!